### PR TITLE
Avoid build require system zlib

### PR DESCRIPTION
**Pull request by yiwu-arbug**
`register_dep("Z")` correctly set the include path for zlib, but fail to set the path to libz.a. RocksDB cmake script expect libz.a be place under ${DEP_Z_ROOT}/lib, but libz-sys place it under ${DEP_Z_ROOT}/build. Fixed it.

Also added a test to make sure possible compression types are all linked.

Signed-off-by: Yi Wu <yiwu@pingcap.com>

### CODE REVIEW

**Comment by siddontang** [comment]
Great.

PTAL @BusyJay

**Comment by siddontang** [comment]
have you tested in MacOSX?

**Comment by siddontang** [comment]
@yiwu-arbug 

If you don't add this, can the newly added test pass the CI?

File: librocksdb_sys/libtitan_sys/build.rs [hunk 1]
```diff
@@ -1,14 +1,20 @@
 extern crate cc;
 extern crate cmake;
 
+use std::env;
+
 fn main() {
+    for e in env::vars() {
```
[the remark this answers is not in the document]
Comment (yiwu-arbug) [reply librocksdb_sys/libtitan_sys/build.rs]: debug only, will remove.

File: librocksdb_sys/build.rs [hunk 2]
```diff
@@ -83,9 +83,10 @@ fn build_rocksdb() -> Build {
     if cfg!(feature = "sse") {
         cfg.define("FORCE_SSE42", "ON");
     }
+    let zlib_dir = env::var("DEP_Z_ROOT").unwrap();
     let dst = cfg
-        .register_dep("Z")
-        .define("WITH_ZLIB", "ON")
+        .cxxflag("-DZLIB")
```
[the remark this answers is not in the document]
Comment (yiwu-arbug) [reply librocksdb_sys/build.rs]: @BusyJay I understand the purpose of `register_dep`, but on my side (minimal image of ubuntu 18.10 without system zlib installed), CMake still complain about zlib not available. I guess it has something to do with the difference between "Z" and "ZLIB"? I also tried `register_dep("ZLIB") and it didn't work. Any idea?

[the remark this answers is not in the document]
Comment (BusyJay) [reply librocksdb_sys/build.rs]: You can see the output file in build directory to see what command is actually being executed. `cmake` can be configured with verbose output, which may be helpful for debugging.

**Comment by siddontang** [comment]
can we go on? seem it blocks for some time.

**Comment by yiwu-arbug** [comment]
> can we go on? seem it blocks for some time.

Need some debugging for why `register_dep("Z")` doesn't work. Will find sometime this week to check.

**Comment by yiwu-arbug** [comment]
Did a bit more investigation. It seems `register_dep("Z")` correctly set the include path for zlib, but fail to set the path to libz.a. RocksDB cmake script expect libz.a be place under ${DEP_Z_ROOT}/lib, but libz-sys place it under ${DEP_Z_ROOT}/build. Fixed it.

@BusyJay mind take another look?

File: librocksdb_sys/build.rs [hunk 3]
```diff
@@ -99,8 +104,10 @@ fn build_rocksdb() -> Build {
         .register_dep("SNAPPY")
         .define("WITH_SNAPPY", "ON")
         .build_target("rocksdb")
+        .very_verbose(true)
```
Comment (BusyJay) [inline librocksdb_sys/build.rs]: Is it necessary?

Comment (yiwu-arbug) [reply librocksdb_sys/build.rs]: I find it helpful for debugging, and it doesn't change `cargo build` output without `-vv`.

**Pull request merged**